src/watchers/indexable-author.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * Author watcher to save the meta data to an Indexable.
+ *
+ * @package Yoast\YoastSEO\Watchers
+ */
+
+namespace Yoast\YoastSEO\Watchers;
+
+use Yoast\YoastSEO\Exceptions\No_Indexable_Found;
+use Yoast\YoastSEO\WordPress\Integration;
+use Yoast\YoastSEO\Yoast_Model;
+use Yoast\YoastSEO\Models\Indexable;
+
+/**
+ * Watches an Author to save the meta information when updated.
+ */
+class Indexable_Author implements Integration {
+
+	/**
+	 * Registers all hooks to WordPress.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'profile_update', array( $this, 'save_meta' ), PHP_INT_MAX, 2 );
+		\add_action( 'deleted_user', array( $this, 'delete_meta' ) );
+	}
+
+	/**
+	 * Deletes user meta.
+	 *
+	 * @param int $user_id User ID to delete the metadata of.
+	 *
+	 * @return void
+	 */
+	public function delete_meta( $user_id ) {
+		try {
+			$indexable = $this->get_indexable( $user_id, false );
+			$indexable->delete();
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+	}
+
+	/**
+	 * Saves user meta.
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return void
+	 */
+	public function save_meta( $user_id ) {
+		try {
+			$indexable = $this->get_indexable( $user_id );
+		} catch ( No_Indexable_Found $exception ) {
+			return;
+		}
+
+		$indexable->permalink = $this->get_permalink( $user_id );
+
+		$meta_data = $this->get_meta_data( $user_id );
+
+		$indexable->title             = $meta_data['wpseo_title'];
+		$indexable->description       = $meta_data['wpseo_metadesc'];
+		$indexable->is_robots_noindex = $this->get_noindex_value( $meta_data['wpseo_noindex_author'] );
+
+		$indexable->save();
+	}
+
+	/**
+	 * Retrieves the meta data for this indexable.
+	 *
+	 * @param int $user_id The user to fetch meta data for.
+	 *
+	 * @return array List of meta entries.
+	 */
+	protected function get_meta_data( $user_id ) {
+		$keys = array(
+			'wpseo_title',
+			'wpseo_metadesc',
+			'wpseo_excludeauthorsitemap',
+		);
+
+		$output = array();
+		foreach ( $keys as $key ) {
+			$output[ $key ] = $this->get_author_meta( $user_id, $key );
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Retrieves the indexable for a user.
+	 *
+	 * @param int  $user_id     The user to retrieve the indexable for.
+	 * @param bool $auto_create Optional. Create the indexable when it does not exist yet.
+	 *
+	 * @return Indexable The indexable for the suppied user ID.
+	 *
+	 * @throws No_Indexable_Found Exception when no Indexable could be found for the supplied user.
+	 */
+	protected function get_indexable( $user_id, $auto_create = true ) {
+		$indexable = Yoast_Model::of_type( 'Indexable' )
+								->where( 'object_id', $user_id )
```

Comment:
Could be worth considering adding a single static method to `Indexable` to find a single `Indexable` based on an `object_id` and `object_type`.